gradle/libs.versions.toml

```diff
@@ -1,8 +1,12 @@
 [versions]
 agp = "8.7.0"
 firebaseBom = "33.4.0"
+gson = "2.8.5"
+jetbrainsKotlinxSerializationJson = "1.6.3"
 kotlin =  "1.8.10"
 coreKtx = "1.13.1"
+kotlinxSerializationJson = "1.3.0"
+kotlinxSerializationJsonVersion = "1.6.0"
```

Comment:
Question: Is there a reason not to used what @Wylwi used in #20 ? I'm talking about `android.util.JsonReader`.

If there is a good reason, I think it's fine. If not, maybe we would want to use the same "json serializer" for all API calls.

Reply:
So:
1. The implementation is much easier to do and to understand the code.
2. I was having problems with the `JsonReader` since for some calls it would just return null when initialized: so this could totally be a problem from my side, but after some digging I found this solution much better.
Now my question to you: Is there any reason you would want me to use the `JsonReader`? If yes I could give it a try at redoing the implementation.

Reply:
No, it's good to me. We may want to use your solution for the OSM API too.